### DISCUSSION TITLE
Paragraph

### DISCUSSION THREAD

**Discussion by jeffreylauwers** [nl-design-system/backlog#212] [Component suggestions]
<img width="480" alt="visueel voorbeeld van het paragraph component" src="https://github.com/nl-design-system/backlog/assets/545207/43d67731-799a-4e31-8164-8f03cd68b75a">

We willen graag meer leren over het **Paragraph** component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook of screenshots. Met deze informatie kunnen we bewijzen dat het component onderdeel moet worden van de NL Design System component library. Al zal dat voor de Paragraph nogal logisch aanvoelen, we willen het component toch dezelfde behandelijk geven als elk ander component.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1706529616876159)

---

## Naam
Paragraph

## Doel
Toont een alinea aan tekst.

## Gerelateerde componenten
[Lead Paragraph](https://github.com/orgs/nl-design-system/discussions/213), [Rich Text Content](https://github.com/orgs/nl-design-system/discussions/289), [Strong](https://github.com/orgs/nl-design-system/discussions/291)

## Richtlijnen
Acceptatiecriteria: [NL Design System Acceptatiecriteria voor Paragraph](https://nldesignsystem.nl/paragraph)

## Onderzoek
[Line Length Readbility - Baymard Institute](https://baymard.com/blog/line-length-readability)
Artikel over een onderzoek naar de ideale regellengte.

[Line length revisited: following the research - Design regression](https://designregression.com/article/line-length-revisited-following-the-research)
Artikel over een onderzoek naar de ideale regellengte.

[Heydon Pickering post op Mastodon](https://front-end.social/@heydon/112053893635249451)
Over het gebruik van px en relatieve waardes voor font size.

**Comment by jeffreylauwers** [comment]
## Amsterdam

- [Storybook](https://amsterdam.github.io/design-system/?path=/docs/components-text-paragraph--docs)
- [GitHub](https://github.com/Amsterdam/design-system/blob/develop/packages/css/src/components/paragraph)
- [Figma - Typography](https://www.figma.com/file/9IGm6IdPUYizBNGsUnueBd/Amsterdam-Design-System?type=design&node-id=9734-12550&mode=design&t=mKlSkXksih2RzYPj-4)
- [Figma - Paragraph](https://www.figma.com/file/9IGm6IdPUYizBNGsUnueBd/Amsterdam-Design-System?type=design&node-id=9734-12550&mode=design&t=Sn0TiS1YZL7izDxg-4)

---

### Screenshots / Varianten

**Storybook**
Default
<img width="1032" alt="Scherm­afbeelding 2024-01-25 om 15 07 59" src="https://github.com/nl-design-system/backlog/assets/545207/a10f8237-911b-4361-98bf-91588894cdb6">

---

### Notities
- Responsive gedrag van fontsize obv clamp en calc in CSS. Hierover heeft Vincent ook verteld tijdens een [Heartbeat](https://youtu.be/Asd3RhL0nQY?si=PT6ESzAwv_0PK7dP).
- Opties voor 'Large' en 'Small'. 'Large' ondergebracht bij [Lead paragraph](https://github.com/orgs/nl-design-system/discussions/213).
- Storybook component biedt de mogelijkheid om de kleur om te keren bij een donkere achtergrond.
- Figma component biedt de mogelijkheid voor spacious en compact weergave.

**Comment by jeffreylauwers** [comment]
## Den Haag

- [Storybook](https://nl-design-system.github.io/denhaag/?path=/docs/css-paragraph--docs)
- [GitHub](https://github.com/nl-design-system/denhaag/tree/main/components/Paragraph)
- [Figma - Huidig](https://www.figma.com/file/JpoY3waVoQGlLQzQXTL9nn/%F0%9F%8C%BB-Den-Haag-Design-System?type=design&node-id=3785-680&mode=design&t=8j01TVaxZWp5LemF-4)
- [Figma - WIP](https://www.figma.com/file/x4RkF6BIdrNZbh7D53NTzB/%F0%9F%92%A0-NLDS---Den-Haag---Bibliotheek?type=design&node-id=450-5082&mode=design&t=x89g6JtciW9NlIJ2-4)

---

### Screenshots / Varianten

**Storybook**
Default
<img width="1040" alt="Scherm­afbeelding 2024-01-25 om 15 10 08" src="https://github.com/nl-design-system/backlog/assets/545207/5869aac8-979e-480d-8f82-19d5f3ea4956">

---

### Notities
Opties voor 'Lead' en 'Detail Small'. 'Lead' ondergebracht bij [Lead paragraph](https://github.com/orgs/nl-design-system/discussions/213).

**Comment by jeffreylauwers** [comment]
## DUO

- [Documentatie website](https://uno.dfront.rijkscloud.nl/#/templates/full-text)
- [Github](https://github.com/dso-toolkit/dso-toolkit/blob/master/packages/dso-toolkit/src/components/paragraph/paragraph.scss)

---

### Screenshots / Varianten

**Documentatie website**

Default
![Scherm­afbeelding 2024-01-25 om 15 28 27](https://github.com/nl-design-system/backlog/assets/545207/6629bdc4-263d-4a2e-ab56-736291d4346c)


---

### Notities
- Geen specifiek component of css class voor de paragraaf. Er is wel een 'Header paragraph' (ondergebracht bij [Lead paragraph](https://github.com/orgs/nl-design-system/discussions/213)) vandaar dat de standaard paragraph hier is opgenomen.

**Comment by jeffreylauwers** [comment]
## Open Formulieren

- [Storybook](https://open-formulieren.github.io/open-forms-sdk/?path=/docs/pure-react-components-body--docs)

---

### Screenshots / Varianten

**Storybook**

Plain
![Scherm­afbeelding 2024-01-25 om 16 18 29](https://github.com/nl-design-system/backlog/assets/545207/4e152c07-e24e-4228-a1fc-fddb5dfbca3b)

---

### Notities
Geen specifiek component of css class voor de paragraaf. Er is wel een 'Body' component met een modifiers onder andere 'Big' en 'Small'. 'Big' (ondergebracht bij [Lead paragraph](https://github.com/orgs/nl-design-system/discussions/213)) vandaar dat de 'Plain' paragraph hier is opgenomen.

**Comment by jeffreylauwers** [comment]
## Open Gemeenten

- [Documentatie website - Standaardpagina](https://componenten.lenteveld.nl/paginatype-standaard)
- [Patternlab](https://patternlab.lenteveld.nl/index.html)

---

### Screenshots / Varianten

**Patternlab**

Bodytext
![Scherm­afbeelding 2024-01-29 om 09 16 50](https://github.com/nl-design-system/backlog/assets/545207/af96d570-b155-436b-ac65-56ce7a0e1108)

---

### Notities
- Optie 'Intro-text' ondergebracht bij [Lead paragraph](https://github.com/orgs/nl-design-system/discussions/213).

**Comment by jeffreylauwers** [comment]
## Rotterdam

- [Storybook](https://nl-design-system.github.io/rotterdam/?path=/docs/css-utrecht-paragraph--docs)
- [Figma](https://www.figma.com/file/ZWSC4gCrOXRUR9UX3aoZ8x/Gemeente-Rotterdam-Design-System?type=design&node-id=415-12136&mode=design&t=0wrujN4GRFXRo3Y7-4)

---

### Screenshots / Varianten

**Storybook**
Default
![Scherm­afbeelding 2024-01-29 om 09 34 02](https://github.com/nl-design-system/backlog/assets/545207/d4c1e164-619d-4a43-8d6b-34399f4b932a)

---

### Notities
- Wordt 'Default' genoemd in Figma.
- In het Figma file wordt vermeld dat een paragraaf over maximaal 8 kolommen van het grid mag lopen.
- Optie 'Lead paragraph' ondergebracht bij [Lead paragraph](https://github.com/orgs/nl-design-system/discussions/213).

**Comment by jeffreylauwers** [comment]
## RVO

- [Storybook - Demo content pagina](https://nl-design-system.github.io/rvo/?path=/story/demo-pagina-s-algemeen-content--default)
- [Storybook - Text helpers](https://nl-design-system.github.io/rvo/?path=/docs/componenten-text-helpers--documentatie)
- [GitHub](https://github.com/nl-design-system/rvo/tree/master/components/paragraph)
- [Figma](https://www.figma.com/file/Sj6myBL1Fvot5M1qGxzvEo/ROOS-(RVO-Design-System)?type=design&node-id=1161-19173&mode=design&t=VeVaoqcI7PaLdZIv-4)

---

### Screenshots / Varianten

**Storybook**
Paragraph Medium
![Scherm­afbeelding 2024-01-29 om 11 00 05](https://github.com/nl-design-system/backlog/assets/545207/76488b98-c2de-40a5-9e39-36ff1315d979)

---

### Notities
- Bij Demo contentpagina en Figma opties voor grootte: Large (rvo-paragraph--lg), Medium (rvo-paragraph--md) en Small (rvo-paragraph--sm).
- Optie 'Large' ondergebracht bij [Lead paragraph](https://github.com/orgs/nl-design-system/discussions/213).
- Aanvullende Text helpers opties voor grootte: sm/md/lg/xl.

**Comment by jeffreylauwers** [comment]
## Utrecht

- [Documentatie website](https://nl-design-system.github.io/utrecht/docs/componenten/typografie/paragraaf)
- [Storybook](https://nl-design-system.github.io/utrecht/storybook/?path=/docs/css_css-paragraph--docs)
- [GitHub](https://github.com/nl-design-system/utrecht/tree/main/components/paragraph)
- [Figma](https://www.figma.com/file/UXIHcIurAD8hyoBWx4hDBV/NLDS---Gemeente-Utrecht---Bibliotheek?type=design&node-id=450-5082&mode=design&t=zjOKLNlBEPHmFua4-4)

---

### Screenshots / Varianten

**Storybook**
Default
![Scherm­afbeelding 2024-01-29 om 12 48 43](https://github.com/nl-design-system/backlog/assets/545207/2ddbdbff-725e-46d6-8b3e-2256c0396bb7)

---

### Notities
- Documentatie website bevat uitgebreide richtlijnen.
- Opties voor 'Lead' en 'Small print'. 'Lead' ondergebracht bij [Lead paragraph](https://github.com/orgs/nl-design-system/discussions/213).

**Comment by jeffreylauwers** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[Paragraph op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=paragraph)

**Comment by jiromaykin** [comment]
Wellicht overbodige aanvulling maar behalve een standaard paragraaf heeft OIP/open-inwoner nog 'modifiers' dus daar maken we onze eigen afwijkende styling, en soms eigen design-tokens van.
Het betreft alleen verschillen in **styling**, dus niet in structuur, als verschillend van de standaard 'utrecht' paragraaf, als modifiers hebben wij (nu nog): `--compact,` `--small`, `--muted`, `--centered`
Maar we hebben ook allerlei unieke paragrafen binnen grotere 'molecule/organism' componenten.
Zoals de kleinere paragrafen binnen tabellen e.a. overzichten of error meldingen in de document-upload component, of de paragraaf in een Tab kolom zoals `<p class="tab-column__paragraph">`.

Het is nu nog aan verandering onderhevig, maar hier is alvast nieuwe CSS te zien:
https://github.com/maykinmedia/open-inwoner/pull/1215/files#diff-e90221354759ce160863b0106bfb76069bb4121659733cf775bb37a9b9ad44ea

Als gemeenten hun eigen design-tokens in ons project zouden laden, in de toekomst, dan zijn de afwijkende OIP tokens/stylingen wel nog via de gegenereerde CSS te customizen. Maar alle componenten overerven i.e.g. de body font-stijlen dus die veranderen altijd wel mee.

**Reply by Robbert** [reply]
@jiromaykin Heel interessant!

Deze `compact` variant die direct onder een `fieldset > legend` staat: deze paragraph zou misschien zijn speciale look kunnen krijgen als 'ie in een form-field-description staat? Is er een standaard styling voor "helper tekst"? Of verschilt de styling van "helper tekst" per comonent? https://github.com/maykinmedia/open-inwoner/pull/1215/files#diff-81e4cb63c2dc557c11c4b2d61fd5021b651d52838f2bacf059f21966d7a0294bR24

**Reply by jiromaykin** [reply]
Die '`help_text`' is eigenlijk een Django specifieke term die idd. per component verschilt en meestal in de Admin gebruikt wordt.
Er is wel wat voor te zeggen om minder modifiers te gebruiken en de styling op de parent componenten te zetten ja (als dat is wat je bedoelt). Maar sommige styling zou ik liever 'project overkoepelend' willen eigenlijk, zoals `--muted` of `--small`, zodat de paragrafen herbruikbaar zijn.

**Comment by jeffreylauwers** [comment]
## ✨ Dit component is nu Community ✨

Help je mee hem door de Candidate stappen te krijgen?
[Paragraph op het Candidate bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=paragraph)

**Comment by jeffreylauwers** [comment]
## ✨ Dit component is nu Candidate ✨

Help je mee hem door de Hall of Fame stappen te krijgen?
[Paragraph op het Hall of Fame bord](https://github.com/orgs/nl-design-system/projects/30/views/1?filterQuery=paragraph)
